src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -226,17 +227,37 @@ private function parseDefaults(array &$content, $file)
         return $defaults;
     }
 
+    private function isUnderscoredParamValid($content, $name, $file)
+    {
+        if (!isset($content['services'][$name])) {
+            return false;
+        }
+
+        if (!is_array($underscoreParam = $content['services'][$name])) {
+            throw new InvalidArgumentException(sprintf('Service "%s" key must be an array, "%s" given in "%s".', $name, gettype($underscoreParam), $file));
+        }
+
+        if (isset($underscoreParam['alias']) || isset($underscoreParam['class']) || isset($underscoreParam['factory'])) {
+            @trigger_error(sprintf('Giving a service the "%s" name is deprecated since Symfony 3.3 and will be forbidden in 4.0. Rename your service.', $name), E_USER_DEPRECATED);
```

Comment:
I'm mixed on this deprecation at general the service id level.
But I'm :+1: for deprecating such id **when using the YamlFileLoader**